src/types/mod.rs

```diff
@@ -76,34 +65,21 @@ impl Hash for Op {
         self.arity.hash(state);
     }
 }
-impl Name for Op {
-    fn name(&self) -> Name_ {
-        self.name.clone()
-    }
-}
```

Comment:
Hmm... why take away access to the `name`?

Reply:
I think I was getting a bit eager with deleting the `Name` trait — I will put these methods back!

Reply:
(this is back in my code review commit — it moved to the body of `impl Operator`)